tests/test_indexes.py

```diff
@@ -1036,9 +1029,9 @@ def test_timedelta_div() -> None:
     delta = dt.timedelta(1)
 
     check(assert_type(index / delta, "pd.Index[float]"), pd.Index, float)
-    check(assert_type(index / 1, "TimedeltaIndex"), pd.Index, pd.Timedelta)
+    check(assert_type(index / 1, "pd.TimedeltaIndex"), pd.TimedeltaIndex, pd.Timedelta)
     check(assert_type(index // delta, "pd.Index[int]"), pd.Index, np.longlong)
-    check(assert_type(index // 1, "TimedeltaIndex"), pd.Index, pd.Timedelta)
+    check(assert_type(index // 1, "pd.TimedeltaIndex"), pd.TimedeltaIndex, pd.Timedelta)
 
     check(assert_type(delta / index, "pd.Index[float]"), pd.Index, float)
     check(assert_type(delta // index, "pd.Index[int]"), pd.Index, np.longlong)
```

Comment:
Should add tests for division by lists of floats/ints and for the invalid usage (similar to what you did for `Series`)

Reply:
I added tests except for two cases, see https://github.com/pandas-dev/pandas/issues/54554